openff/nagl_models/tests/test_dynamic_fetch.py

```diff
@@ -83,60 +45,94 @@ def hide_cache():
         SocketBlockedError,
     ):
-        get_model.cache_clear()
+        get_model(
+            "my_favorite_model.pt",
+            doi="10.5072/zenodo.278300",
+            file_hash="127eb0b9512f22546f8b455582bcd85b2521866d32b86d231fee26d4771b1d81",
+        )
 
-        get_model("openff-gnn-am1bcc-0.1.0-rc.3.pt")
+    # Ensure that the file can actually be fetched
+    enable_socket()
+    get_model(
+        "my_favorite_model.pt",
+        doi="10.5072/zenodo.278300",
+    )
 
+    # Ensure that, once fetched, the file can be gotten without accessing the internet.
+    disable_socket()
+    # Ensure that cached files can be accessed when no optional arguments are provided
+    get_model(
+        "my_favorite_model.pt",
+    )
 
-def test_file_exists_in_cache_without_internet(monkeypatch):
-    # since tests can run in different orders, make sure the file exists already
-    with monkeypatch.context() as m:
-        m.setattr(
-            urllib.request,
-            "urlretrieve",
-            mocked_urlretrieve,
-        )
-        m.setattr(
-            openff.nagl_models._dynamic_fetch,
-            "get_release_metadata",
-            mocked_get_release_metadata,
-        )
+    # Ensure that cached files can be accessed when only doi is provided
+    get_model(
+        "my_favorite_model.pt",
+        doi="10.5072/zenodo.278300",
+    )
 
-        assert get_model("openff-gnn-am1bcc-0.1.0-rc.3.pt")
+    # Ensure that cached files can be accessed when all optional arguments are provided
+    get_model(
+        "my_favorite_model.pt",
+        doi="10.5072/zenodo.278300",
+        file_hash="127eb0b9512f22546f8b455582bcd85b2521866d32b86d231fee26d4771b1d81",
+    )
 
-        disable_socket()
+    # Ensure that cached files can be accessed when only hash is provided
+    get_model(
+        "my_favorite_model.pt",
+        file_hash="127eb0b9512f22546f8b455582bcd85b2521866d32b86d231fee26d4771b1d81",
+    )
+
+    # Ensure that cached files raise hash comparison errors
+    with pytest.raises(HashComparisonFailedException):
+        get_model(
+            "my_favorite_model.pt",
+            doi="10.5072/zenodo.278300",
+            file_hash="wrong_hash",
+        )
 
-        get_model("openff-gnn-am1bcc-0.1.0-rc.3.pt")
 
 
-def test_error_on_missing_file(monkeypatch):
-    with (
-        pytest.raises(
+#
+def test_error_on_missing_file():
+    with pytest.raises(
             FileNotFoundError,
-            match="Could not find asset with name 'FOOBAR",
-        ),
-        monkeypatch.context() as m,
-    ):
-        m.setattr(
-            urllib.request,
-            "urlretrieve",
-            mocked_urlretrieve,
-        )
-        m.setattr(
-            openff.nagl_models._dynamic_fetch,
-            "get_release_metadata",
-            mocked_get_release_metadata,
-        )
+            match="Could not find asset with name 'FOOBAR"):
+        get_model("FOOBAR.pt")
+
+def test_error_on_bad_file_suffix():
```

Comment:
Since this behavior should not reach out to the internet, it would be nice to have this test run with and without the socket turned off. Same with `test_error_on_bad_file_suffix`.